src/data/data.cc

```diff
@@ -2,15 +2,16 @@
  * Copyright 2015-2019 by Contributors
  * \file data.cc
  */
-#include <xgboost/data.h>
-#include <xgboost/logging.h>
-#include <xgboost/build_config.h>
 #include <dmlc/registry.h>
 #include <cstring>
 
-#include "./sparse_page_writer.h"
-#include "./simple_dmatrix.h"
-#include "./simple_csr_source.h"
+#include "xgboost/data.h"
+#include "xgboost/logging.h"
+#include "xgboost/version_config.h"
+#include "sparse_page_writer.h"
+#include "simple_dmatrix.h"
+#include "simple_csr_source.h"
+
```

Comment:
This re-arrangement of headers can be avoided by including `version_config.h` in `base.h`.

Reply:
@hcho3 It might be better to include what you use.  The linter checks for this convention for standard headers, but not for others.